server/api/crud/runtimes/nuclio/function.py

```diff
@@ -347,13 +347,15 @@ def _compile_function_config(
     _set_function_replicas(function, nuclio_spec)
     _set_misc_specs(function, nuclio_spec)
 
+    is_nuclio_image_set = function.spec.config.get("spec.image") is not None
+
     # if the user code is given explicitly or from a source, we need to set the handler and relevant attributes
     if (
         function.spec.base_spec
         or function.spec.build.functionSourceCode
         or function.spec.build.source
         or function.kind == mlrun.runtimes.RuntimeKinds.serving  # serving can be empty
-    ):
+    ) and not is_nuclio_image_set:
```

Comment:
can you explain this? The else clause seems to be weird. why would we want to build file if a nuclio image is set?

Reply:
You're right, I don't want to enter the "else" clause.
The actual meaning was to not call `_set_source_code_and_handler`, so I'll move the condition there.